### PR TITLE
[ASCII-2048] use telemetry component inside the autodiscovery component

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Updates the autodiscovery component to use the telemetry component instead of the `pkg/telemetry` 

### Motivation

We are moving away from using globals to declare our telemetry metrics 

### Describe how to test/QA your changes

- Ensure the telemetry metrics are being emitted as before

### CODE REVIEW

**Review by AlexandreYang — APPROVED** [review]

**Review by NouemanKHAL — APPROVED** [review]

**Review by sblumenthal — APPROVED** [review]

**Review by hush-hush — APPROVED** [review]

**Review by gh123man — APPROVED** [review]
AML owned files LGTM

File: cmd/agent/common/autodiscovery.go [hunk 1]
```diff
@@ -40,8 +40,11 @@ var (
 
 func setupAutoDiscovery(confSearchPaths []string, wmeta workloadmeta.Component, ac autodiscovery.Component) {
 	providers.InitConfigFilesReader(confSearchPaths)
+
+	acTelemetryStore := ac.GetTelemetryStore()
+
 	ac.AddConfigProvider(
-		providers.NewFileConfigProvider(),
+		providers.NewFileConfigProvider(acTelemetryStore),
```
Comment (clamoriniere) [inline cmd/agent/common/autodiscovery.go:47]: dummy question why the FileConfigProvider need to have now the TelemetryStore?

**Review by clamoriniere — APPROVED** [review]

**Pull request merged**